### PR TITLE
refactor: move functions getting prerequisite to feature domain

### PR DESCRIPTION
**Pull request by masaaania**
- This PR moves functions getting prerequisites from the api-gateway domain to the feature domain since those functions are used in domains other than the api-gateway.

### CODE REVIEW

File: pkg/feature/domain/evaluation.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func TopologicalSort(features []*featureproto.Feature) ([]*featureproto.Feature,
 		for _, p := range f.Prerequisites {
 			pf, ok := mapFeatures[p.FeatureId]
 			if !ok {
-				return errFeatureNotFound
+				return ErrFeatureNotFound
```
Comment (ono-max) [inline pkg/feature/domain/evaluation.go:120]: Do you have a reason why we need to change from `errFeatureNotFound` to `ErrFeatureNotFound`?

Comment (masaaania) [reply pkg/feature/domain/evaluation.go:120]: There is no need, thx!

Comment (masaaania) [reply pkg/feature/domain/evaluation.go]: @ono-max 
I fixed it 👍

**Review by ono-max — APPROVED** [review]
👍

**Pull request merged**